### DISCUSSION TITLE
Beginner Friendly Installation Guide From Scratch (Minimal Setup to get it working)

### DISCUSSION THREAD

**Discussion by MrWildanMD** [Keyfactor/ejbca-ce#368] [Q&A] [answered]
Okay what make me decide to make this discussion is, i'm trying to set-up on my local machine without docker to build ejbca from scratch, following along the official documentation quickstart the required steps (exactly the same), but when it comes firing it up, it getting me authorization denied, and even figuring out myself, i dont find something that could fixing it, revoke -> regenerate -> clear browser cache -> install p12 on browser again -> try entering private tab, but still throwing the same result, maybe there more neat documentation or atleast getting it worked. Thank you!
![Screenshot_2023-09-27_23_04_02](https://github.com/Keyfactor/ejbca-ce/assets/60354284/a9b50fb3-a6c9-40aa-9767-9162a4bec6b9)
![image](https://github.com/Keyfactor/ejbca-ce/assets/60354284/8430d581-6074-4046-9e8d-fb2b2f550e2d)

**Comment by primetomas** [comment]
You don't give much information when guide you followed, and what versions etc. The log looks good, but the "No client certificate presented", on port 8443 should kind of not be possible on a standard install as port 8443 would be configured to require client certificate authentication, so the TLS connection should typically not be established at all if you don't get a prompt for you client certificate.

Exactly what installation instructions are you following? What WildFly version, etc?

**Reply by MrWildanMD** [reply]
Okay, i'm sorry not providing the current version used
current using Latest v8.0.0 EJBCA
appserver: WildFly v26
database: MySQL latest
os: kali linux latest version

so this is my step procedured
1. https://doc.primekey.com/ejbca/ejbca-installation/installation-prerequisites checking if pre-req is satisfied
2. Cloning the repo and following configuration along from there(exactly doing same configuration)  https://doc.primekey.com/ejbca/ejbca-installation/managing-ejbca-configurations
3.  After done configuring the EJBCA, then i followed guide to creating the database and make sure the user is created properly
![Screenshot_2023-09-27_12_48_51](https://github.com/Keyfactor/ejbca-ce/assets/60354284/e55ec566-d7e9-4e23-80cd-fd0bdfa91153)
4. Following WildFly configuration until the audit logging https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26
5. Deploying the EJBCA to the WildFly https://doc.primekey.com/ejbca/ejbca-installation/deploying-ejbca
![Screenshot_2023-09-27_20_25_05](https://github.com/Keyfactor/ejbca-ce/assets/60354284/441504df-3de4-483d-a61e-4ca911229053)
6. Finally installing it https://doc.primekey.com/ejbca/ejbca-installation/installing-ejbca/installing-ejbca-as-a-ca-with-a-management-ca
7. Installing the p12 to browser then access the adminweb give authorization denied

kindly thanks for your response 🙏

**Comment by MrWildanMD** [comment]
Did i miss some configuration on the WildFly side..since i'm only following the guide until the configuring the audit log

**Reply by primetomas** [reply, marked as answer]
It would be the HTTPS configuration, https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-HTTP(S)Configuration
I recommend to use the section for 3-port separation, it's easier. Then ignore the 2-port separation section.

**Reply by MrWildanMD** [reply]
So after following till the audit logging then continue to 3 port separation right?
alright gotta give it shot

**Comment by MrWildanMD** [comment]
Thanks, it working perfectly now,,but in Parrot OS,,,would be great if pointed more which configuration needed (the required ejbca configuration is good)....and now i got CRL warning in the adminweb....is that fine?

**Reply by primetomas** [reply]
Great to hear. With the documentation improvement, can you be specific what would have helped you? Being clear about 3-post separation vs 2-port? (with 3-port being the recommended one?)
